flaky test `test_matrix_multi_user_roaming`

In my recent PRs I see an teardown error in `test_matrix_multi_user_roaming`.

As it happens in two independent PRs I think it's not related to them.

- https://circleci.com/gh/raiden-network/raiden/85374#tests/containers/13
- https://circleci.com/gh/raiden-network/raiden/85382#tests/containers/13
- https://circleci.com/gh/raiden-network/raiden/85533#tests/containers/13
- https://circleci.com/gh/raiden-network/raiden/85439#tests/containers/13


```
_ ERROR at teardown of test_matrix_multi_user_roaming[matrix-private_rooms0-6-3] _

tp = <class 'RuntimeError'>, value = None, tb = None

    def reraise(tp, value, tb=None):
        try:
            if value is None:
                value = tp()
            if value.__traceback__ is not tb:
                raise value.with_traceback(tb)
>           raise value

../venv-3.7-LINUX/lib/python3.7/site-packages/six.py:693: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../venv-3.7-LINUX/lib/python3.7/site-packages/six.py:693: in reraise
    raise value
../venv-3.7-LINUX/lib/python3.7/site-packages/six.py:693: in reraise
    raise value
raiden/tests/integration/fixtures/transport.py:88: in matrix_transports
    transport.get()
src/gevent/greenlet.py:633: in gevent._greenlet.Greenlet.get
    ???
src/gevent/greenlet.py:312: in gevent._greenlet.Greenlet._raise_exception
    ???
../venv-3.7-LINUX/lib/python3.7/site-packages/gevent/_compat.py:47: in reraise
    raise value.with_traceback(tb)
src/gevent/greenlet.py:716: in gevent._greenlet.Greenlet.run
    ???
raiden/network/transport/matrix/transport.py:457: in _run
    self.stop()  # ensure cleanup and wait on subtasks
raiden/network/transport/matrix/transport.py:481: in stop
    self._client.stop()  # stop sync_thread, wait on client's greenlets
raiden/network/transport/matrix/client.py:316: in stop
    self.stop_listener_thread()
raiden/network/transport/matrix/client.py:306: in stop_listener_thread
    {self._handle_thread}, timeout=SHUTDOWN_TIMEOUT, raise_error=True
src/gevent/greenlet.py:849: in gevent._greenlet.joinall
    ???
src/gevent/greenlet.py:865: in gevent._greenlet.joinall
    ???
src/gevent/greenlet.py:312: in gevent._greenlet.Greenlet._raise_exception
    ???
../venv-3.7-LINUX/lib/python3.7/site-packages/gevent/_compat.py:47: in reraise
    raise value.with_traceback(tb)
src/gevent/greenlet.py:716: in gevent._greenlet.Greenlet.run
    ???
raiden/network/transport/matrix/client.py:440: in _handle_response
    self.call(callback, presence_update)
raiden/network/transport/matrix/client.py:401: in call
    return callback(*args, **kwargs)
raiden/network/transport/matrix/utils.py:263: in _presence_listener
    self.refresh_address_presence(address)
raiden/network/transport/matrix/utils.py:200: in refresh_address_presence
    self._fetch_user_presence(uid) for uid in self._address_to_userids[address]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   composite_presence = {
        self._fetch_user_presence(uid) for uid in self._address_to_userids[address]
    }
E   RuntimeError: Set changed size during iteration

--- a/raiden/network/transport/matrix/utils.py
+++ b/raiden/network/transport/matrix/utils.py
@@ -197,7 +197,7 @@ class UserAddressManager:
         should **not** generally be used.
         """
         composite_presence = {
-            self._fetch_user_presence(uid) for uid in self._address_to_userids[address]
+            self._fetch_user_presence(uid) for uid in self._address_to_userids[address].copy()
         }
 
         # Iterate over UserPresence in definition order (most to least online) and pick

--- a/raiden/tests/integration/network/transport/test_matrix_transport.py
+++ b/raiden/tests/integration/network/transport/test_matrix_transport.py
@@ -923,7 +923,6 @@ def test_matrix_user_roaming(matrix_transports):
     assert ping_pong_message_success(transport0, transport1)
 
 
-@pytest.mark.skip(reason="flaky, see https://github.com/raiden-network/raiden/issues/5127")
 @pytest.mark.parametrize("matrix_server_count", [3])
 @pytest.mark.parametrize("number_of_transports", [6])
 def test_matrix_multi_user_roaming(matrix_transports):

